services/CurlingEventService.js

```diff
@@ -368,6 +368,100 @@ class CurlingEventService {
       throw error;
     }
   }
+
+  async createTeam(name, affiliation, note) {
+    const values = [name, affiliation, note];
+    try {
+      const data = await this.#pool
+        .query(Queries.CREATE_TEAM, values);
+      return data.rows;
+    }
+    catch (err) {
+      if (err.message.includes('affiliation')) {
+        throw Exceptions.insertionException('Invalid affiliation/organization id.');
+      }
+    }
+  }
+
+  async updateTeam(id, name, affiliation, note) {
+    if (affiliation === null || affiliation === undefined) {
+      const events = await this.#pool.query(Queries.GET_FRIENDLY_EVENTS_BY_TEAM, [id]);
+      if (events.rowCount > 0) {
+        throw Exceptions.updateException('Team is a part of a "friendly" curling event');
```

Comment:
you want to check if any of the events are part of a 'friendly' event. So you'll have to go through the rows